src/System.Collections.Specialized/src/System/Collections/Specialized/NameValueCollection.cs

```diff
@@ -262,14 +262,14 @@ internal virtual bool InternalHasKeys()
         ///    <para>Adds an entry with the specified name and value into the
         ///    <see cref='System.Collections.Specialized.NameValueCollection'/>.</para>
         /// </devdoc>
-        public virtual void Add(string name, string value)
+        public virtual void Add(string? name, string? value)
         {
             if (IsReadOnly)
                 throw new NotSupportedException(SR.CollectionReadOnly);
 
             InvalidateCachedArrays();
 
-            ArrayList values = (ArrayList)BaseGet(name);
+            ArrayList? values = BaseGet(name) as ArrayList;
```

Comment:
You don't need to / shouldn't change the kind of cast employed here, just make it:
```C#
ArrayList? values = (ArrayList?)BaseGet(name);
```